File: program/print-number-to-words-in-indian-numbering-system/PrintNumberToWordsInIndianNumberingSystem.js
function getOnes(number) {
  const ones = ['', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine'];
  return ones[number] || '';
}

function getTeens(number) {
  const teens = ['ten', 'eleven', 'twelve', 'thirteen', 'fourteen', 'fifteen', 'sixteen', 'seventeen', 'eighteen', 'nineteen'];
  return teens[number] || '';
}

function getTens(number) {
  const tens = ['', '', 'twenty', 'thirty', 'forty', 'fifty', 'sixty', 'seventy', 'eighty', 'ninety'];
  return tens[number] || '';
}

function getHundreds(number) {
  if (number > 0 && number < 10) {
    return getOnes(number);
  }
  if (number >= 10 && number < 20) {
    return getTeens(number % 10);
  }
  if (number >= 20 && number < 100) {
    return `${getTens(Math.floor(number / 10))} ${getOnes(number % 10)}`;
  }
  return '';
}

function numberToWords(input) {
  const digits = Number(parseInt(input, 10));
  const result = [];

  if (digits === 0) {
    result.push('zero');
  } else if (digits === 1) {
    result.push('one');
  } else {
    const crores = Math.floor(digits / 10000000) % 100;
    if (crores > 0) {
      result.push(`${getHundreds(crores)} crore`);
    }

    const lakhs = Math.floor(digits / 100000) % 100;
    if (lakhs > 0) {
      result.push(`${getHundreds(lakhs)} lakh`);
    }

    const thousands = Math.floor(digits / 1000) % 100;
    if (thousands > 0) {
      result.push(`${getHundreds(thousands)} thousand`);
    }

    const hundreds = Math.floor((digits % 1000) / 100);
    if (hundreds > 0 && hundreds < 10) {
      result.push(`${getOnes(hundreds)} hundred`);
    }

    const tens = digits % 100;
    if (tens > 0) {
      if (digits > 100) result.push('and');
      result.push(`${getHundreds(tens)}`);
    }
  }

  return [...result].join(' ').split(/\s/).filter((e) => e).map((e) => e.substr(0, 1)
          .toUpperCase() + e.substr(1)).join(' ');
}


console.log(numberToWords(123456789));

File: program/program/print-number-to-words-in-indian-numbering-system/print_number_to_words_in_indian_numbering_system.pl
use strict;
use warnings;

our @words = ("", "One", "Two", "Three", "Four", "Five", "Six", "Seven",
              "Eight", "Nine", "Ten", "Eleven", "Twelve", "Thirteen", "Fourteen",
              "Fiveteen", "Sixteen", "Seventeen", "Eighteen", "Nineteen");

our @tens = ("", "", "Twenty", "Thirty", "Forty", "Fifty",
             "Sixty", "Seventy", "Eighty", "Ninety");

our @face_val = ("Crore", "Lakh", "Thousand", "Hundred");

our @divisor = (10000000, 100000, 1000, 100, 1);

sub parse {
    my $n = shift;

    if ($n < 20) {
        print $words[$n];
    } else {
        print $tens[$n / 10], " ", $words[$n % 10];
    }
}

sub main {
    print "Enter number: ";
    my $n = <STDIN>;
    chomp($n);

    if ($n == 0) {
        print "Zero\n";
        return;
    }

    for (my $i = 0; $i < 5; $i++) {
        my $div = $divisor[$i];
        if (int($n / $div) != 0) {
            parse(int($n / $div));
            if ($i != 4) {
                print " $face_val[$i] ";
            }
        }
        $n %= $div;
    }

    print "\n";
}

main();
